src/main/java/org/opensearch/flowframework/workflow/WorkflowProcessSorter.java

```diff
@@ -141,7 +142,7 @@ public List<ProcessNode> sortProcessNodes(Workflow workflow, String workflowId)
      * @throws Exception if validation fails
      */
     public void validate(List<ProcessNode> processNodes) throws Exception {
-        WorkflowValidator validator = WorkflowValidator.parse("mappings/workflow-steps.json");
+        WorkflowValidator validator = readWorkflowValidator(processNodes.get(0).id());
```

Comment:
This id is just the workflow step ID of the first node, not the workflow ID that method is expecting, so If we don't yet have a workflow ID I'd suggest just passing "new workflow" or something similar here.